C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -217,25 +215,39 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
 
                     try
                     {
-                        SlackAPI slack = await this.GetAPIAsync(turnContext.Activity);
-                        ChatPostMessageResult result = null;
+                        SlackTaskClient slack = await this.GetAPIAsync(turnContext.Activity);
+                        ChatPostMessageResult result = new ChatPostMessageResult();
 
                         if (message.ephemeral)
                         {
-                            // result = await slack.Chat // TODO
+                            Console.WriteLine("Post Ephemeral:", message);
+                            result = await slack.PostEphemeralMessageAsync(message.Channel, message.Text, message.User);
                         }
                         else
                         {
-
+                            Console.WriteLine("Post Message:", message);
+                            result = await slack.PostMessageAsync(message.Channel, message.Text);
                         }
 
-                        // if (result.Ok) // TODO
+                        if ((result as dynamic).Ok)
```

Comment:
Does `ChatPostMessageResult` not have an ok or similar property?
Since we know the exact class that `result` is, there is no use to use `dynamic`.

Reply:
Good point

Reply:
I think that the problem here is _ChatPostMessageResult_ must extend another _interface_ from the _Inumadia/SlackAPI_. Note that this is the current implementation in _TypeScript_:

![image](https://user-images.githubusercontent.com/43762887/58639930-64f28500-82ce-11e9-8096-5e6c160f9017.png)

![image](https://user-images.githubusercontent.com/43762887/58639955-7176dd80-82ce-11e9-9ed8-8acc39091cbf.png)